controlplane/internal/controllers/rke2controlplane_controller.go

```diff
@@ -309,6 +310,22 @@ func (r *RKE2ControlPlaneReconciler) updateStatus(ctx context.Context, rcp *cont
 		rcp.Status.Initialized = true
 	}
 
+	availableCPMachines := readyMachines.Filter(collections.Not(collections.HasUnhealthyCondition))
+	validIpAddresses := []string{}
+	for _, machine := range availableCPMachines {
+		ipAddress, err := getIPAddress(*machine)
+		if err != nil {
+			break
```

Comment:
Would it be beneficial to have a debug log of the error?

Reply:
In my opinion, having a CP Machine without an IP address is normal. So, it will happen frequently. What seems more relevant is if at the end of the loop, the list of valid IP addresses is empty, which I account for in a specific `if` statement.
Let me know what you think!